### PR TITLE
#11 logic and tests done

### PR DESCRIPTION
**Pull request by JavierGarna**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -196,6 +196,66 @@ describe("GET /api/reviews", () => {
...
+        });
+    });
+    test.only("404: should return a not found message when given a non-existent category", () => {
+        return request(app)
+          .get("/api/reviews?category=somethingElse")
+          .expect(404)
+          .then((response) => {
+            expect(response.body.msg).toBe("not found");
+        });
+    });
```
Comment (sirKevlar) [inline __tests__/app.test.js:274]: It's possible to query a valid category which has no reviews... In this case we need another test: 200 returns empty array

File: models/reviews.models.js [hunk 2]
```diff
@@ -32,15 +32,60 @@ exports.fetchPatchReview = (review_id, inc_votes) => {
...
+    //     })
+    // } else {
+    //     return getReviews.then((response) => {
+    //         return response.rows;
+    //     })
+    // }
+
+    return getReviews.then((response) => {
+        if (!response.rows.length) {
+            return db.query(`SELECT * FROM categories WHERE slug = $1`, array).then((response) => {
```
Comment (k-huggs) [inline models/reviews.models.js]: This can be handled in a check categories function then use a promise.all, rather than nesting this

**Review by k-huggs — COMMENTED** [review]
Hi Javier, you can utilise another function to check if things exist i.e. categories or review_id's. That way you don't have to nest your queries

**Review by emilyb93 — APPROVED** [review]
